python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -339,9 +339,9 @@ def transceiver_qsfp_default_mode_4x10(self) -> bool:
 
     @cached_property
     def event_monitor(self) -> dict | None:
-        """event_monitor set based on event_monitor data-model. TODO: add to schema."""
-        if get(self._hostvars, "event_monitor") is True:
-            return {"enabled": "true"}
+        """event_monitor set based on event_monitor data-model."""
+        if get(self._hostvars, "event_monitor.enabled") is True:
+            return {"enabled": True}
         return None
```

Comment:
This should be refactored to use the classes instead:
```suggestion
        """event_monitor set based on event_monitor data-model."""
        return self.inputs.event_monitor.as_dict() or None
```

Reply:
this should be ._as_dict() :)
fixed!